packages/scoped-elements/README.md

```diff
@@ -0,0 +1,232 @@
+how custom elements are defined, registering and scoping their tag names if it is
+necessary, avoiding the name collision problem.
+
+## Installation
+
+```bash
+npm i --save @open-wc/scoped-elements
+```
+
+## How it works
+
+`createScopedHtml` is going to define the custom element classes and creates a
+unique scoped tag for each of them if needed. In case those classes were
+previously defined then it's going to return the previous defined tag.
+
+Then, the `html` function provided by the `createScopedHtml` method is going to
+transform the template literal into another one replacing the tags used by the
+user by the ones defined by the custom elements. Finally, the transformed
+template literal is going to be processed by `lit-html`.
+
+`<my-button>${this.text}</my-button>` --> `<my-button-3>${this.text}</my-button-3>`
+
+## How to use it
+
+1. Import the `createScopedHtml` from `@open-wc/scoped-elements`.
+
+   ```js
+   import { createScopedHtml } from '@open-wc/scoped-elements';
+   ```
+
+2. Import the classes of the components you want to use.
+
+   ```js
+   import MyButton from './MyButton.js';
+   import MyPanel from './MyPanel.js';
+   ```
+
+3. Create the `html` function defining the tags you want to use for the
+   components.
+
+   ```js
+   const html = createScopedHtml({
+     'my-button': MyButton,
+     'my-panel': MyPanel,
+   });
+   ```
+
+4. Use your components in your html.
+
+   ```js
+   render() {
+     return html`
+       <my-panel class="my-panel">
+         <my-button>${this.text}</my-button>
+       </my-panel}>
+     `;
+   }
+   ```
+
+### Complete example
+
+```js
+import { css, LitElement } from 'lit-element';
+import { createScopedHtml } from '@open-wc/scoped-elements';
+import MyButton from './MyButton.js';
+import MyPanel from './MyPanel.js';
+
+const html = createScopedHtml({
+  'my-button': MyButton, // <-- bind the element with a tag in your html
+  'my-panel': MyPanel,
+});
+
+export default class MyElement extends LitElement {
+  static get styles() {
+    return css`
+      .my-panel {
+        padding: 10px;
+        background-color: grey;
+      }
+    `;
+  }
+
+  static get properties() {
+    return {
+      text: String,
+    };
+  }
+
+  render() {
+    return html`
+      <my-panel class="my-panel">
+        <my-button>${this.text}</my-button>
+      </my-panel}>
+    `;
+  }
+}
+```
+
+### What if I need to use the component tag?
+
+Sometimes we need to use the component tag name, for example, if we want to do a
+`querySelector`. In that case there is a `getTags` function available that receives
+a collection of component classes and returns a collection with the tag names
+used in their definitions.
+
+```js
+import { getTags } from '@open-wc/scoped-elements';
+import { ComponentA } from './ComponentA.js';
+import { ComponentB } from './ComponentB.js';
+
+const [componentA, componentB] = getTags([ComponentA, ComponentB]);
```

Comment:
I am not sure if we wanna expose getTags 🤔 

what happens if we use both?
```js
const html = createScopedHtml({
  'my-button': MyButton, // <-- bind the element with a tag in your html
  'my-panel': MyPanel,
});

const [tagMyButton, tagMyPanel] = getTags([MyButton, MyPanel]);
```

are we sure that it is always the exact same thing? and even if... the API seems confusing to me 🙈 

if we wanna expose it - how about
```js
const { html, tags } = createScopedHtml({
  'my-button': MyButton, // <-- bind the element with a tag in your html
  'my-panel': MyPanel,
});

document.querySelector(tags['my-button']);
```

Reply:
Yes, it´s the same thing. I think we must expose the tags in some way but I like your proposal using `createdScopedHtml`.

I´m going to do the code changes.

Reply:
tags don't need to get exposed in a complicated way you can simply return from createScopedHtml the WeakMap or in your case map and get the ElementName via the Class. also you don't need to expose the original class names in your config.

Reply:
Can we hold off on this for the first release? We need to see how this works with the build plugin.

Reply:
agree let's hold of with that for now but make it an object so we can later add "something" if needed?

```js
const { html } = createScopedHtml(...)
```

Reply:
I already did the modification returning the tags in the object. Could we hold it with that or you prefer a rollback to the previous one?

Reply:
I think we should go with `const html = createScopedHtml(...)` to start off, and evaluate if the other things are needed later and when we know if and how we want to deal with compiling parts of this away. The listed use cases in the docs can be solved with css classes, ids or attributes.

Reply:
indeed...

Reply:
Yes, that's true. I'm going to rollback to that version without the `getTags` function.

Reply:
Sorry I dont mean to devaluate your work, I want nothing more than this feature to succeed !

Reply:
No worries, I understand you. And it´s true that users can use css classes, ids or attributes to cover that functionality. Let´s see if it´s really needed in the future. I really appreciate your comments.

The less code the better.

Reply:
I just updated the PR. Let's see if it's ok as a start point now.